loky/backend/context.py

```diff
@@ -8,93 +8,49 @@
 #  * Create a context ensuring loky uses only objects that are compatible
 #  * Add LokyContext to the list of context of multiprocessing so loky can be
 #    used with multiprocessing.set_start_method
-#  * Add some compat function for python2.7 and 3.3.
 #
-from __future__ import division
-
 import os
 import sys
+import math
 import subprocess
 import traceback
 import warnings
 import multiprocessing as mp
+from multiprocessing import get_context as mp_get_context
+from multiprocessing.context import BaseContext
 
 
 from .process import LokyProcess, LokyInitMainProcess
 
-START_METHODS = ['loky', 'loky_init_main']
+START_METHODS = ['loky', 'loky_init_main', 'spawn']
+if sys.platform != 'win32':
+    START_METHODS += ['fork', 'forkserver']
+
 _DEFAULT_START_METHOD = None
 
 # Cache for the number of physical cores to avoid repeating subprocess calls.
 # It should not change during the lifetime of the program.
 physical_cores_cache = None
 
-if sys.version_info[:2] >= (3, 4):
-    from multiprocessing import get_context as mp_get_context
-    from multiprocessing.context import assert_spawning, set_spawning_popen
-    from multiprocessing.context import get_spawning_popen, BaseContext
-
-    START_METHODS += ['spawn']
-    if sys.platform != 'win32':
-        START_METHODS += ['fork', 'forkserver']
-
-    def get_context(method=None):
-        # Try to overload the default context
-        method = method or _DEFAULT_START_METHOD or "loky"
-        if method == "fork":
-            # If 'fork' is explicitly requested, warn user about potential
-            # issues.
-            warnings.warn("`fork` start method should not be used with "
-                          "`loky` as it does not respect POSIX. Try using "
-                          "`spawn` or `loky` instead.", UserWarning)
-        try:
-            context = mp_get_context(method)
-        except ValueError:
-            raise ValueError(
-                f"Unknown context '{method}'. Value should be in "
-                f"{START_METHODS}."
-            )
-
-        return context
-
-else:
-    if sys.platform != 'win32':
-        import threading
-        # Mechanism to check that the current thread is spawning a process
-        _tls = threading.local()
-        popen_attr = 'spawning_popen'
-    else:
-        from multiprocessing.forking import Popen
-        _tls = Popen._tls
-        popen_attr = 'process_handle'
-
-    BaseContext = object
-
-    def get_spawning_popen():
-        return getattr(_tls, popen_attr, None)
-
-    def set_spawning_popen(popen):
-        setattr(_tls, popen_attr, popen)
```

Comment:
I believe the above was removed too eagerly:

```python-traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/runner/work/1/s/loky/__init__.py", line 19, in <module>
    from .backend.reduction import set_loky_pickler
  File "/Users/runner/work/1/s/loky/backend/reduction.py", line 119, in <module>
    from ._posix_reduction import _mk_inheritable  # noqa: F401
  File "/Users/runner/work/1/s/loky/backend/_posix_reduction.py", line 15, in <module>
    from .context import get_spawning_popen
ImportError: cannot import name 'get_spawning_popen' from 'loky.backend.context' (/Users/runner/work/1/s/loky/backend/context.py)
```

Reply:
This PR needed the "reduction.py" PR to be merged first

Reply:
Ah ok.